presto-main/src/main/java/com/facebook/presto/operator/FileFragmentResultCacheManager.java

```diff
@@ -88,7 +90,9 @@ public FileFragmentResultCacheManager(
 
         this.baseDirectory = Paths.get(cacheConfig.getBaseDirectory());
         this.maxInFlightBytes = cacheConfig.getMaxInFlightSize().toBytes();
-        this.pagesSerde = new PagesSerdeFactory(blockEncodingSerde, cacheConfig.isBlockEncodingCompressionEnabled()).createPagesSerde();
+        this.maxSinglePagesBytes = cacheConfig.getMaxSinglePagesSize().toBytes();
+        this.pagesSerdeFactory = new PagesSerdeFactory(blockEncodingSerde, cacheConfig.isBlockEncodingCompressionEnabled());
```

Comment:
Is this change still needed? Maybe it would be sufficient to keep `pagesSerde` only?

Reply:
the `pagesSerde` is not thread safe ,when writing data use a share buffer,and read is noe use the share buffer